src/math/MathUtils.js

```diff
@@ -54,6 +54,22 @@ const MathUtils = {
 
 	},
 
+	// https://www.gamedev.net/tutorials/programming/general-and-gameplay-programming/inverse-lerp-a-super-useful-yet-often-overlooked-function-r5230/
+
+	inverseLerp: function ( x, y, value ) {
+
+		if ( x !== y ) {
+
+			return MathUtils.clamp( ( value - x ) / ( y - x ), 0, 1 );
```

Comment:
why are we using clamp? neither the current [`.lerp()`](https://github.com/mrdoob/three.js/blob/02f4dff06e26304ce67b98f0777786f3ca9428a6/src/math/MathUtils.js#L61) or [`.mapLinear()`](https://github.com/mrdoob/three.js/blob/02f4dff06e26304ce67b98f0777786f3ca9428a6/src/math/MathUtils.js#L53) implementations uses clamp, it's better to leave it to the user.

Unity is different because it clamps function by default, for example Unity has `LerpUnclamped` because their `Lerp` implementation is already clamping by default.

Reply:
+1

Reply:
I've updated the PR although I think it would be more safe to use `MathUtils.clamp()`.

Reply:
@Mugen87 the issue is that sometimes the unclamped behaviour is needed (talking from experience), so you'd have to provide it somehow to the users, and the options are the following:

1. Provide an alternative unclamped method, such as `LerpUnclamped` in Unity
2. Provide a toggle such as in [canvas-sketch-util's mapRange function](https://github.com/mattdesl/canvas-sketch-util/blob/master/docs/math.md#n--maprangevalue-inputmin-inputmax-ouptutmin-outputmax-clamp--false)
3. Leave it up to the user to wrap it in a `clamp()` function, which is the current three.js approach

I think the current three.js approach is the easier to manage.